### PR TITLE
Add an standalone training script

### PR DESCRIPTION
**Pull request by juansebashr**

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+    )
+    crawling_keyword_examples = (
+        subprocess.check_output(
+            [
+                "find",
+                os.path.join(data_dir, "ours", "crawling-keyword-examples"),
+                "-type",
+                "f",
+                "-name",
+                "*.wav",
```
Comment (charlielito) [inline train.py]: add comment of what that data is about

Comment (charlielito) [reply train.py]: But these data is in BigQuery, why do you need to use gsutil list?

File: train.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
+
```
Comment (charlielito) [inline train.py:127]: what is this?

File: train.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                os.path.join(data_dir, "ours", "away-negative-keyword-examples"),
+                "-type",
+                "f",
+                "-name",
+                "*.wav",
+            ]
+        )
+        .decode("utf-8")
+        .split("\n")
+    )
```
Comment (charlielito) [inline train.py]: same, add comment about the data

File: train.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+    # embedding.trainable = False
+
+    # Prepare data
+    negative_files_str = [str(path) for path in negative_files]
+
+    # Split the list of files into train, validation, and test sets
+    train_negatives, test_negatives = train_test_split(
+        negative_files_str, test_size=0.1, random_state=42
+    )
+
```
Comment (charlielito) [inline train.py:251]: :( add a TODO that we should split this better to avoid data leakage

File: train.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    if toy:
+        num_epochs = 1
+        num_batches = 1
+        backprop_into_embedding = False
+    else:
+        num_epochs = 10
+        num_batches = 2
+        backprop_into_embedding = True
+
```
Comment (charlielito) [inline train.py]: make these arguments of the main functon

File: train.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+        num_epochs=num_epochs,
+        num_batches=num_batches,
+        batch_size=64,
+        primary_lr=0.001,
+        backprop_into_embedding=backprop_into_embedding,
+        embedding_lr=0.0005,
+        model_settings=model_settings,
+        base_model_path=model_dir,
+        base_model_output="dense_2",
+        UNKNOWN_PERCENTAGE=50.0,
```
Comment (charlielito) [inline train.py]: make this also argumment of script

File: train.py [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+        continue_training=False,
+    )
+
+    model.save(model_dir.parent / "finetuned_model")
+
+    test_spectrograms = np.array([file2spec(model_settings, f) for f in test_samples])
+    # fetch softmax predictions from the finetuned model:
+    # (class 0: silence/background noise, class 1: unknown keyword, class 2: target)
+    predictions = model.predict(test_spectrograms)
+    categorical_predictions = np.argmax(predictions, axis=1)
```
Comment (charlielito) [inline train.py]: what? is that comment right? Do we only have 2 classes??

File: multilingual_kws/embedding/transfer_learning.py [hunk 8]
```diff
@@ -28,36 +28,40 @@ def transfer_learn(
...
+    else:
+        xfer = tf.keras.models.Model(
+            name="TransferLearnedModel",
+            inputs=base_model.inputs,
+            outputs=base_model.get_layer(name=base_model_output).output,
+        )
+        xfer.trainable = False
+
+        # dont use softmax unless losses from_logits=False
+        CATEGORIES = 2  # other + target_keyword
```
Comment (charlielito) [inline multilingual_kws/embedding/transfer_learning.py]: categories should be an agument!

**Review by charlielito — APPROVED** [review]

File: train.py (hunk 4 above)
Comment (juansebashr) [reply train.py:251]: I added it a few lines after this

File: train.py (hunk 7 above)
Comment (juansebashr) [reply train.py]: AH shit, my bad, I copy it from the notebook without noticing

File: train.py (hunk 5 above)
Comment (juansebashr) [reply train.py]: But, make sense taking into account that I'm also having "toy" as argument?

Comment (charlielito) [reply train.py]: Yes, toy override them

Comment (juansebashr) [reply train.py]: aye aye

File: train.py (hunk 1 above)
Comment (juansebashr) [reply train.py]: The ones that are on this directory are not all the images that we recorded; first I took max 15 per person, and also It turns out that all the recordings that we did are not in the correct format, so I used the script I left on Tamagotchi repo tu convert them

File: multilingual_kws/embedding/transfer_learning.py (hunk 8 above)
Comment (juansebashr) [reply multilingual_kws/embedding/transfer_learning.py]: Done

File: train.py (hunk 7 above)
Comment (juansebashr) [reply train.py]: Done

File: train.py (hunk 6 above)
Comment (juansebashr) [reply train.py]: Done

File: train.py (hunk 5 above)
Comment (juansebashr) [reply train.py]: Done, I left default the toy paramters and left a comment of recommended parameters with full training

File: train.py (hunk 4 above)
Comment (juansebashr) [reply train.py:251]: Done

File: train.py (hunk 3 above)
Comment (juansebashr) [reply train.py]: Done

File: train.py (hunk 2 above)
Comment (juansebashr) [reply train.py:127]: Done

File: train.py (hunk 1 above)
Comment (juansebashr) [reply train.py]: Done

Comment (charlielito) [reply train.py]: -.-

**Pull request merged**